File: 2018/go/day01/day01.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func Sum(freqs []int) int {
	sum := 0
	for _, f := range freqs {
		sum += f
	}
	return sum
}

func Calibrate(freqs []int) int {
	seen := make(map[int]bool)
	current := 0
	seen[current] = true
	i := 0

	for {
		idx := i % len(freqs)
		f := freqs[idx]
		current += f

		if seen[current] {
			return current
		}

		i++
		seen[current] = true
	}
}

func main() {
	input, err := ioutil.ReadFile("/Users/hhh/JungleGym/advent_of_code/2018/inputs/day1.in")
	if err != nil {
		panic(err)
	}
	freqs := formatInput(string(input))

	fmt.Printf("Day 1 Part 1: %+v\n", Sum(freqs))
	fmt.Printf("Day 1 Part 2: %+v\n", Calibrate(freqs))
}

func formatInput(input string) []int {
	freq_str := strings.Split(input, "\n")
	out := []int{}

	for _, s := range freq_str {
		i, err := strconv.Atoi(s)
		if err == nil {
			out = append(out, i)
		}
	}
	return out
}

File: 2021/day01.rb
depths = File.readlines("/Users/hhh/JungleGym/advent_of_code/2021/inputs/day01.in").map { |x| x.strip.to_i }

pt1 = depths.each_cons(2).count { |(x,y)| y - x > 0 }
p "Part 1: #{pt1}"

pt2 = depths.each_cons(3).map { |x| x.sum }.each_cons(2).count { |(x,y)| y - x > 0 }
p "Part 2: #{pt2}"
